build.gradle

```diff
@@ -10,31 +10,25 @@ buildscript {
         classpath "io.github.gradle-nexus:publish-plugin:2.0.0"
         classpath("org.asciidoctor:asciidoctor-gradle-jvm:4.0.2")
     }
-
-    configurations.configureEach {
-        resolutionStrategy.eachDependency { DependencyResolveDetails details ->
-            // This is a workaround for dependencies exposing a different version of Groovy
-            // than the one used by Gradle. This causes issues with the Groovy compiler.
-            if (details.requested.group == 'org.apache.groovy' || details.requested.group == 'org.codehaus.groovy') {
-                details.useTarget(group: "org.codehaus.groovy", name: details.requested.name, version: GroovySystem.version)
-                details.because "Use Groovy 3.0.x library shipped with Gradle"
-            }
-        }
-    }
-}
-
-repositories {
-    maven { url "https://repo.grails.org/grails/core" }
 }
 
 group = "org.grails"
 version project.projectVersion
 
+apply plugin: 'java-library'
+apply plugin: 'groovy'
+
 ext {
-    commonBuild = 'https://raw.githubusercontent.com/grails/grails-common-build/gradle7'
+
+    isSnapshot = project.projectVersion.endsWith('-SNAPSHOT')
+    isReleaseVersion = !isSnapshot
 }
 
-allprojects  {
+ext."signing.keyId" = project.hasProperty("signing.keyId") ?: System.getenv('SIGNING_KEY')
+ext."signing.secretKeyRingFile" = project.hasProperty("signing.secretKeyRingFile") ?: "${System.properties['user.home']}${File.separator}.gnupg${File.separator}secring.gpg"
+ext."signing.password" = project.hasProperty("signing.password") ?: System.getenv('SIGNING_PASSPHRASE')
```

Comment:
Change `project.hasProperty('x')` to `project.findProperty('x')`

Reply:
Updated